cmd/sync/sync.go

```diff
@@ -91,25 +116,12 @@ func New() *cobra.Command {
 	cmd.Flags().Var(&f.output, "output", "type of output format")
 
 	cmd.RunE = func(cmd *cobra.Command, args []string) error {
-		var opts *sync.SyncOptions
-		var err error
-
-		//
 		// To be uncommented and used once our VS Code extension is bundle aware.
 		// Until then, this could interfere with extension usage where a `databricks.yml` file is present.
 		// See https://github.com/databricks/cli/pull/207.
 		//
-		// b := bundle.GetOrNil(cmd.Context())
-		// if b != nil {
-		// 	// Run initialize phase to make sure paths are set.
-		// 	err = bundle.Apply(cmd.Context(), b, phases.Initialize())
-		// 	if err != nil {
-		// 		return err
-		// 	}
-		// 	opts, err = syncOptionsFromBundle(cmd, args, b)
-		// } else {
-		opts, err = f.syncOptionsFromArgs(cmd, args)
-		// }
+		opts, err := f.syncOptions(cmd, args, phases.Initialize())
```

Comment:
why not just pass `phases.Initialise` directly to `err = bundle.Apply(cmd.Context(), b, phases.Initialise())` ? Then you don't need to pass it in your function as parameter

Reply:
Initialize makes API calls which are difficult to unwrangle and mock in this case (sincle bundle is also parsed by function). This makes it easier to test.